python/dgl/dataloading/limited_edge.py

```diff
@@ -0,0 +1,113 @@
+from typing import Generic
+import functools
+
+import numpy as np
+import torch
+import dgl
+from dgl.dataloading.dataloader import _TensorizedDatasetIter
+
+
+class LimitedEdgeDataloader(dgl.dataloading.NodeDataLoader):
+    """Limited Edge Dataloader class.
+
+    For the normal dataloader, each batch have a same batch size, which may cause the workload unbanlance
+    problem when some batches contains high degree nodes. The limited edge dataloader can support different
+    batch sizes among different batches. Users can parse two configures: max_node and max_edge, which means
+    the maximum number of nodes in a batch and the maximum number of edges in a batch. The limited edge 
+    dataloader can provide the largest batch which follows the rule.
+    """
+    def __init__(self, g, nids, sampler, max_node, max_edge, prefix_sum_in_degrees=None, \
```

Comment:
1. You could make `max_edge` (still think `max_degree` is better) a mandatory argument while making `max_node` an optional one. You could give it a default value (e.g., 0) meaning that the dataloader will not control minibatches by num nodes.
2. I see. Then it's ok to leave it here. Just need to update the doc.
3. I see.

Reply:
1. I think set `max_node` as `None` to represent is a possible solution. But in my opinion, `max_edge` is to control the maximum number of edges inside a Graph (also known as block), It is strange to describe this item as "degree"
, since we do not say the degree of a block.
2. Okay, I will add it again. I think we can not allow `prefix_sum_in_degrees` if the `shuffle` is True.

Reply:
1. It depends on the sampler user provides. For full neighbor sampler, you are right. But users may provide subgraph sampler, etc.
2. Then remember to have a validity check.

Reply:
Okay I changed it to `max_degree`.